File: messageGo/grpc/methodData/start_service.go
package methodData

import (
	"game-message-core/proto"
)

// 向manager service 请求启动服务 request
type StartServiceInput struct {
	ServiceType     proto.ServiceType         `json:"serviceType"`
	SceneSerSubType proto.SceneServiceSubType `json:"sceneSerSubType"`
	OwnerId         int64                     `json:"ownerId"`
	MapId           int32                     `json:"mapId"`
}

type StartServiceOutput struct {
	Success bool   `json:"success"`
	ErrMsg  string `json:"errMsg"`
}

File: messageDotNet/grpc/MethodData/StartService.cs
using System;
using UnityEngine;

/// <summary>
/// 向manager service 请求启动服务 request
/// </summary>
[Serializable]
public class StartServiceInput
{
    public GameMessageCore.ServiceType ServiceType;
    public GameMessageCore.SceneServiceSubType SceneSerSubType;
    public long OwnerId;
    public int MapId;
    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }
}


/// <summary>
/// 向manager service 请求启动服务  response
/// </summary>
[Serializable]
public class StartServiceOutput
{
    public bool Success;
    public string ErrMsg;

    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }

}
